### PR TITLE
[tests] use correct storage size

### PR DESCRIPTION
**Pull request by olafhering**
## Types of change

- [x ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/filesystem/test/TestFileFactory.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ TEST_F(TestFileFactory, Read)
         str = StringUtils::Format("%02X ", buf[i]);
         std::cout << str;
       }
-      while (i++ < sizeof(buf))
+      while (i++ < (ssize_t)sizeof(buf))
```
Comment (pkerling) [inline xbmc/filesystem/test/TestFileFactory.cpp]:
```suggestion
      while (i++ < static_cast<ssize_t> (sizeof(buf)))
```

File: xbmc/filesystem/test/TestZipFile.cpp [hunk 2]
```diff
@@ -192,7 +192,7 @@ TEST_F(TestZipFile, CorruptedFile)
       str = StringUtils::Format("%02X ", buf[i]);
       std::cout << str;
     }
-    while (i++ < sizeof(buf))
+    while (i++ < (ssize_t)sizeof(buf))
```
Comment (pkerling) [inline xbmc/filesystem/test/TestZipFile.cpp]:
```suggestion
      while (i++ < static_cast<ssize_t> (sizeof(buf)))
```

**Comment by olafhering** [comment]
Updated with static_cast. Android builds are broken, looks unrelated to my change.

**Comment by pkerling** [comment]
More `size_t` is always welcome. Thanks. Let's try android once more.

**Comment by pkerling** [comment]
jenkins build and merge

**Review by yol — APPROVED** [review]

**Pull request merged**